src/early-types.lisp

```diff
@@ -164,6 +164,13 @@ Signals an error if FOREIGN-TYPE is undefined."))
   (:documentation
    "Return the size in bytes of a foreign type."))
 
+(define-compiler-macro foreign-type-size (&whole form foreign-type)
+  ;; Don't pass &environment to CONSTANTP or MACROEXPAND as LOAD-TIME-VALUE is
+  ;; executed in a null lexical environment.
+  (if (constantp (macroexpand foreign-type))
+      `(load-time-value (locally (declare (notinline foreign-type-size)) ,form))
```

Comment:
It's better to use CONSTANT-FORM-VALUE here. L-T-V inhibits futher optimisations, such as (foreign-alloc (+ <constant> (foreign-type-size :foo)))

Reply:
Fixed, thanks :+1:

Reply:
Did you undo the change to use CONSTANT-FORM-VALUE ?

Reply:
I don't *think* so...? Unless I did that in process of reordering the commits.